Create user API

zo1lu opened the pull request:
Hi @cwpeng

Here's the testing link
http://52.63.186.90:3000/
http://52.63.186.90:3000/attraction/10

by _zzzoe

zo1lu committed:
Create 3 user API for member account management and authentication

-Create enroll route and login route
-Create authenticate membership route
-Create token with jwt when validated

zo1lu committed:
Revise return error message

zo1lu committed:
Create login and registration feature

-Create popup form
-Create login and register function
-Create logout function
-Create login autheticate request when load page

zo1lu committed:
Refactore frontend files

-Destructure origin .html into components
-Create shared css file for shared components which were used across multi page
-Create component using js CustomElementRegistry

zo1lu committed:
Fix popup box gradient header deco

zo1lu committed:
Make sure the parameter type passing to database is string

zo1lu committed:
Fix error message color to red

cwpeng reviewed (changes requested):
1. Show error message without replacing existing text.
2. Show sign in form first when user click on 登入/註冊.

zo1lu committed:
Modify login popup box display

-Show login box when click button in navbar
-Show result message below submit button than replace switching-link

zo1lu committed:
Make login box show each time clicking the button in navbar

zo1lu commented:
Hi @cwpeng 

Modification completed!

Testing link:
http://52.63.186.90:3000/
http://52.63.186.90:3000/attraction/10

by _zzzoe

cwpeng reviewed (approved):
Good. Done.

zo1lu committed:
Add booking API

-Create bookings table in database
-APIs for get booking list, create booking, delete booking
-Authenticate first in each booking API

zo1lu committed:
Create booking page

zo1lu committed:
Fix bug

zo1lu committed:
Use js module to share function across pages

zo1lu committed:
Create tour-booking request

zo1lu committed:
Create get-booking-list request

-Booking page RWD
-Show booking list with return data
-Add delete-booking function
